src/robot/reporting/jsbuildingcontext.py

```diff
@@ -68,6 +72,13 @@ def create_link_target(self, msg):
         id = self._top_level_strings.add(msg.parent.id)
         self._msg_links[self._link_key(msg)] = id
 
+    def check_auto_expansion(self, kw):
+        self._auto_expand_matcher.check(kw)
```

Comment:
This might be premature optimization, but I'm slightly worried about the performance if that `.check(kw)` is called with all keywords. There can be a huge amount of keywords and calling matchers internally in that `check()` method is not free. It might be better to set `self._auto_expand_matcher to `None` if this functionality isn't enabled and then here have:
```python
if self._auto_expand_matcher is not None:
    # ...
```

Reply:
refactoring done: now _expand_matcher is set to None if functionality is not needed avoiding calls to .check methods